osu.Game/Rulesets/Mods/Mod.cs

```diff
@@ -147,15 +147,16 @@ public void CopyFrom(Mod source)
                 var targetBindable = (IBindable)prop.GetValue(this);
                 var sourceBindable = (IBindable)prop.GetValue(source);
 
-                // we only care about changes that have been made away from defaults.
-                if (!sourceBindable.IsDefault)
+                // we only care about changes that have been made away from defaults, or if source is default but target is not.
+                if (!sourceBindable.IsDefault || !targetBindable.IsDefault)
```

Comment:
I'm not feeling OK with this check. It looks like a "just fix this" sort of check, by which I mean it fixes the test failure, but it has no actual underlying meaning (why - *in general* - should customised bindables always take in values from source, even if they're defaults? I'm having a hard time justifying that change in my head.)

I would probably prefer adding a `overwriteCustomSettings` bool param to `CopyFrom` and specifying that to `true` in the `ModSection` usage, but I'm not sure everyone will agree to that.

Reply:
Well, this is a `CopyFrom` method, it should copy properties from the source mod into its bindables, so it feels somewhat sensible to me that the copy doesn't occur only when both the source setting and the target setting are unchanged. (for preserving states like `userTrackedSettings` from being changed)

To be fair though it's still kinda haphazard and I agree with you, but I can't find any other sensible way to fix this, so at least this would fix the issue for now until the `OverridableBindable` path or another more resilient path comes and gets rid of all this logic.

And as for your preferred alternative fix, seems like something that would work, but then would there be a reason not to copy to a customized mod at all? it wasn't done at the point of adding this check due to no existence of a `CopyFrom` where the target mod may be already customized.

Reply:
Honestly now that I think about it I'm not even convinced my proposal makes sense anymore. I originally could not and still cannot grasp this logic because I still consider difficulty adjust broken by design because of it trying to keep beatmap default values (that change as the beatmap changes) in the bindable defaults. It'll probably break the same way it did when you removed the guard altogether.

Reply:
Going back to this: if difficulty adjust is the only mod that needed the `!source.IsDefault` check in the first place, then maybe it should be local to that mod, at least as a temporary measure? It's virtual as is anyways.

Reply:
Though I think that would cause the issue to reappear when copying from a default difficulty adjust mod to a customized difficulty adjust mod..

Reply:
Right. Possibly it should be doing something like `userChangedSetting != userChangedSettings[target]` before setting the latter to check if there is a need to copy.

Reply:
difficulty adjust is not the only mod. this is intended to work for all mods with settings (the user choice should be retained if non-default).